core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java

```diff
@@ -89,18 +100,151 @@ public ViewElementDefinition getElement(final String group) {
+            globalEntities = Collections.unmodifiableList(globalEntities);
+        }
+
+        if (null != globalEdges) {
+            globalEdges = Collections.unmodifiableList(globalEdges);
+        }
+    }
+
+    /**
+     * Copies all the global element definitions into the individual element definitions.
+     * The global element definitions will then be set to null
+     */
+    public void expandGlobalDefinitions() {
+        if (null != globalEntities && !globalEntities.isEmpty()) {
+            setEntities(expandGlobalDefinitions(getEntities(), getEntityGroups(), globalEntities, false));
+            globalEntities = null;
+        }
+
+        if (null != globalEdges && !globalEdges.isEmpty()) {
+            setEdges(expandGlobalDefinitions(getEdges(), getEdgeGroups(), globalEdges, false));
+            globalEdges = null;
+        }
+
+        if (null != globalElements && !globalElements.isEmpty()) {
+            setEntities(expandGlobalDefinitions(getEntities(), getEntityGroups(), globalElements, true));
+            setEdges(expandGlobalDefinitions(getEdges(), getEdgeGroups(), globalElements, true));
+            globalElements = null;
+        }
+    }
+
+    private Map<String, ViewElementDefinition> expandGlobalDefinitions(
+            final Map<String, ViewElementDefinition> elements,
+            final Set<String> groups,
+            final List<GlobalViewElementDefinition> globalElements,
+            final boolean skipMissingGroups) {
+
+        final Map<String, ViewElementDefinition> newElements = new LinkedHashMap<>();
+        for (final GlobalViewElementDefinition globalElement : globalElements) {
+            final Set<String> globalGroups;
+            if (null != globalElement.groups) {
+                globalGroups = new HashSet<>(globalElement.groups);
+                final boolean hasMissingGroups = globalGroups.retainAll(groups);
+                if (hasMissingGroups) {
+                    if (!skipMissingGroups) {
+                        final Set<String> missingGroups = new HashSet<>(globalElement.groups);
+                        missingGroups.removeAll(groups);
+                        throw new IllegalArgumentException("A global element definition is invalid, these groups do not exist: " + missingGroups);
+                    }
+                }
+            } else {
+                globalGroups = groups;
+            }
+            for (final String group : globalGroups) {
+                final ViewElementDefinition.Builder builder = new ViewElementDefinition.Builder();
+                if (newElements.containsKey(group)) {
+                    builder.merge(newElements.get(group));
+                }
+                builder.merge(globalElement.clone());
+                newElements.put(group, builder.build());
+            }
+        }
+
+        if (null != elements) {
+            for (final Map.Entry<String, ViewElementDefinition> entry : elements.entrySet()) {
```

Comment:
For interest, Java 8 lets us use lambdas now:

<pre><code>
elements.forEach((k,v) -> {
                if (newElements.containsKey(k)) {
                    newElements.put(k, new ViewElementDefinition.Builder()
                            .merge(newElements.get(k))
                            .merge(v)
                            .build());
                } else {
                    newElements.put(k, v);
                }
            });</code></pre>